libraries/chain/database/block_tasks/process_funds.cpp

```diff
@@ -113,38 +59,67 @@ void process_funds::on_apply(block_task_context& ctx)
     asset original_fund_reward = asset(0, SP_SYMBOL);
     if (fund_budget_service.is_exists())
     {
-        original_fund_reward += fund_budget_management_algorithm(fund_budget_service, dgp_service)
-                                    .allocate_cash(fund_budget_service.get());
+        original_fund_reward += fund_budget_service.allocate_cash(fund_budget_service.get());
     }
 
     distribute_reward(ctx, original_fund_reward); // distribute SP
 
-    asset advertising_budgets_reward = asset(0, SCORUM_SYMBOL);
+    advertising_auction auction(dgp_service, adv_property_svc, post_budget_service, banner_budget_service);
+    auction.run_round();
 
-    advertising_budgets_reward += allocate_advertising_cash(
-        post_budget_service, dgp_service, account_service, advertising_property_service.get().auction_post_coefficients,
-        budget_type::post, ctx);
-    advertising_budgets_reward += allocate_advertising_cash(
-        banner_budget_service, dgp_service, account_service,
-        advertising_property_service.get().auction_banner_coefficients, budget_type::banner, ctx);
+    auto adv_budgets_reward = asset(0, SCORUM_SYMBOL);
```

Comment:
lets move adv budget logic in separate function which will return single const value `adv_budgets_reward`.